### PR TITLE
Adds G80 ammo and magazine to military itemgroups and migrate 12mm Ammo.

### PR DESCRIPTION
**Pull request by Mecares**
Since the Gun itself is in a military itemgroup it makes sense to add
the magazine and ammo also to military itemgroups.
Adds the ammo to ammo_rifle_milspec and the magazine to mags_milspec.

/edit: Also i moved _all_ of the damage from being a gun parameter to the ammo to be more consistent with other weapons. Having most of the damage be a guns parameter would only make sense if the ammo could be used significantly less effective by other weapons.

/edit2: Also moved 12mm Ammo to its own file and minor changes to it as suggested by Mugling.

### CODE REVIEW

File: data/json/itemgroups/ammo.json [hunk 1]
```diff
@@ -166,7 +166,8 @@
       [ "8mm_fmj", 30 ],
       [ "8mm_hvp", 10 ],
       [ "8mm_inc", 20 ],
-      [ "8mm_jhp", 10 ]
+      [ "8mm_jhp", 10 ],
+      [ "12mm", 10 ]
```
Comment (mugling) [inline data/json/itemgroups/ammo.json]: You need alphabetical ordering. A higher spawn rate would be acceptable, say 40?

File: data/json/itemgroups/magazines.json [hunk 2]
```diff
@@ -23,7 +23,8 @@
       ["m107a1mag", 50],
       ["m2010mag", 10 ],
       ["scarhmag", 50],
-      ["scarhbigmag", 20]
+      ["scarhbigmag", 20],
+      ["hk_g80mag", 5]
```
Comment (mugling) [inline data/json/itemgroups/magazines.json]: Again ordering and higher spawn rate, 40?

File: data/json/items/gun/12mm.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
     "material": "steel",
     "ammo": "12mm",
     "skill": "rifle",
-    "ranged_damage": 45,
+    "ranged_damage": 5,
```
Comment (mugling) [inline data/json/items/gun/12mm.json]: A good idea but you should push all of the `damage` to the ammo

Comment (Mecares) [inline data/json/items/gun/12mm.json]: Done.

File: data/json/items/ammo.json [hunk 4]
```diff
@@ -422,7 +422,7 @@
         "volume" : 1,
         "weight" : 11,
         "ammo_type" : "12mm",
-        "damage" : 45,
```
Comment (mugling) [inline data/json/items/ammo.json]: You could consider migrating the `ammo` to the correct subdir and linting it as well

Comment (Mecares) [inline data/json/items/ammo.json]: I thought this was left intentional in ammo.json to avoid a file for just one item.

Comment (mugling) [inline data/json/items/ammo.json]: No, we were originally moving ammo types all in one go (update base items, magazines, ammo and spawns) but that seems to have fallen by the wayside and led to partially migrated example such as `12mm`

Comment (Mecares) [inline data/json/items/ammo.json]: Ok

**Comment by mugling** [comment]
You want to try the web linter for this

**Comment by Mecares** [comment]
Where can i find it?

**Comment by mugling** [comment]
http://dev.narc.ro/cataclysm/format.html

**Comment by Mecares** [comment]
Thank you. Looks very useful.

File: data/json/items/ammo/12mm.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "symbol": "=",
+    "color": "blue",
+    "count": 20,
+    "stack_size": 40,
+    "ammo_type": "12mm",
+    "range": 70,
+    "damage": 50,
+    "pierce": 40,
+    "dispersion": 45,
+    "recoil": 105
```
Comment (Rivet-the-Zombie) [inline data/json/items/ammo/12mm.json]: Why not 110 or 103?

There's nothing wrong with it being 105.

Comment (mugling) [inline data/json/items/ammo/12mm.json]: > There's nothing wrong with it being 105.

It's three significant figures which implies a greater degree of precision than is actually present - that last 0.5% makes no difference. It's the same reason we don't allow a dispersion of `45.6`.

So 110 would be equally reasonable but 103 would not.

Lots of little differences also make comparisons for equality and statistical binning more tricky but that's a side issue.

A lot of the aiming and recoil updates involve some fairly hefty math so we should avoid making the values more awkward then necessary for no tangible gain

Comment (Mecares) [inline data/json/items/ammo/12mm.json]: Done

**Comment by BorkBorkGoesTheCode** [comment]
>Also i moved all of the damage from being a gun parameter to the ammo to be more consistent with other weapons. Having most of the damage be a guns parameter would only make sense if the ammo could be used significantly less effective by other weapons.

No it does not.

**Comment by Mecares** [comment]
> No it does not.

Care to elaborate what the advantage is of having all damage be a gun parameter?

**Comment by mugling** [comment]
> > No it does not 

> Care to elaborate what the advantage is of having all damage be a gun parameter?

None that I can see, your original approach (and rationale for it) are correct

**Pull request merged**